tests/support/mock.py

```diff
@@ -18,26 +18,32 @@
 import errno
 import fnmatch
 import sys
-
-# By these days, we should blowup if mock is not available
-import mock  # pylint: disable=blacklisted-external-import
-
-# pylint: disable=no-name-in-module,no-member
-from mock import (
-    ANY,
-    DEFAULT,
-    FILTER_DIR,
-    MagicMock,
-    Mock,
-    NonCallableMagicMock,
-    NonCallableMock,
-    PropertyMock,
-    __version__,
-    call,
-    create_autospec,
-    patch,
-    sentinel,
-)
+import importlib
+
+current_version = (sys.version_info.major, sys.version_info.minor)
+
+# Use python-mock for older Python versions that are not longer supported
+# in the standard library.
```

Comment:
This comment is slightly misleading. `python-mock` is a rolling backport of CPython's main branch and can contain features that have not been backported to supported Python versions. Right now, `python-mock-5.1.0` includes `ThreadingMock`. That's a new class that was only added a few months ago and is not in any supported Python release:
```
>>> from unittest.mock import ThreadingMock
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'ThreadingMock' from 'unittest.mock' (/usr/lib64/python3.12/unittest/mock.py)
>>> from mock import ThreadingMock
>>> 
```

Reply:
So if upstream Salt uses (now or in the future) features in `python-mock` not yet forward-ported to `unittest.mock` in stdlib then we would have a problem anyway running the test suite in Tumbleweed, is that correct?

Reply:
Yes, if Salt uses features in `python-mock` that have not been backported from CPython@main to CPython@3.X where 3.X is all versions in TW, we have a problem running the testsuite. This is not very likely to cause big issues, if you look at the history of `unittest.mock`, new mock classes are not added often. And when they are, they cover very specific cases and Salt first needs to adopt them. Let's ignore that for now.